draft-jennings-moq-secure-objects.md

```diff
@@ -460,12 +406,39 @@ having processed a valid ciphertext.  In other words, the decryption
 operation should take the same amount of time regardless of whether decryption
 succeeds or fails.
 
-TODO: how to handle errors deserializing private extensions?
+
+# Header Extensions
+
+## Key ID Extension {#keyid-ext}
+
+Key ID  (Extension Header Type 0x2) is a variable length integer and
+identifies the keying material ( keys, nonces and associated context
+for the MOQT Track) to be used for a given MOQT track.
+
+The Key ID extension is included within Immutable Header extension.
+All objects encoded MUST include the Key ID header extension when
+using this specification for object encryption.
+
+## Private Extension {#pvt-ext}
+
+The Private Extensions (Extension Header Type 0xA) contains a sequence of
+Key-Value-Pairs (see section 1.4.2 {{MoQ-TRANSPORT}}) which are also
+Object Extension Headers of the Object. This extension can be added by the
+Original Publisher and considered part of the Object Payload.
```

Comment:
I think we need to talk more about the what the Private Extensions are part of and where they go both before and after encryption and how they are encoded on th wire a bit more.